lib/spack/spack/cmd/repo.py

```diff
@@ -420,6 +445,30 @@ def repo_set(args):
     tty.msg(f"Updated repo '{namespace}'")
 
 
+def repo_update(args: Any) -> int:
+    config = spack.config.CONFIG
+    scope = args.scope
+
+    if args.namespace:
+        existing: Dict[str, Any] = config.get("repos", default={}, scope=scope)
+        if args.namespace not in existing:
+            raise SpackError(f"{args.namespace} is not a known repository namespace.")
+
+        entry = spack.util.path.canonicalize_path(args.namespace)
+        descriptors = {
+            args.namespace: spack.repo.parse_config_descriptor(
+                args.namespace or "<unnamed>", entry, lock=spack.repo.package_repository_lock()
+            )
+        }
+    else:
+        descriptors = spack.repo.RepoDescriptors.from_config(
+            spack.repo.package_repository_lock(), spack.config.CONFIG
+        )
+
+    for _, descriptor in descriptors.items():
+        descriptor.update(git=spack.util.executable.which("git"))
```

Comment:
We should update the descriptor branch/tag/commit here or pass them down as arguments to update.

Reply:
Done.